Code/Framework/AzQtComponents/AzQtComponents/Tests/FloatToStringConversionTests.cpp

```diff
@@ -64,3 +64,43 @@ TEST(AzQtComponents, FloatToString_TruncateShowThousandsSeparatorOnlyOneDecimal)
     int numDecimalPlaces = 2;
     EXPECT_EQ(AzQtComponents::toString(1000.000, numDecimalPlaces, testLocal, showThousandsSeparator), "1,000.0");
 }
+
+TEST(AzQtComponents, FloatToString_Truncate2DecimalsWithLocale)
+{
+    QLocale testLocal{ QLocale() };
+
+    const bool showThousandsSeparator = false;
+    const int numDecimalPlaces = 2;
+    QString testString = "0" + QString(testLocal.decimalPoint()) + "12";
+    EXPECT_EQ(testString, AzQtComponents::toString(0.1234, numDecimalPlaces, testLocal, showThousandsSeparator));
+}
+
+TEST(AzQtComponents, FloatToString_AllZerosButOneWithLocale)
+{
+    QLocale testLocal{ QLocale() };
+
+    const bool showThousandsSeparator = false;
+    const int numDecimalPlaces = 2;
+    QString testString = "1" + QString(testLocal.decimalPoint()) + "0";
+    EXPECT_EQ(testString, AzQtComponents::toString(1.0000, numDecimalPlaces, testLocal, showThousandsSeparator));
```

Comment:
numDecimalPlaces=2 produces "1.0" instead of "1.00"?

Reply:
No, the rule is that multiple zeroes after the decimal place are truncated to a single one.